pkg/cmd/resolve/options.go

```diff
@@ -0,0 +1,295 @@
+type ResolvedTarget struct {
+	// Garden is the garden where the clusters are hosted.
+	Garden Garden `json:"garden"`
+
+	// Project is the project related to the resolved target. It is optional, hence the omitempty tag.
+	Project *Project `json:"project,omitempty"`
+
+	// Shoot is the shoot cluster related to the resolved target. It is optional, hence the omitempty tag.
+	Shoot *Shoot `json:"shoot,omitempty"`
+
+	// Seed is the seed cluster related to the resolved target. It is optional, hence the omitempty tag.
+	Seed *Seed `json:"seed,omitempty"`
+}
+
+// newOptions returns initialized options.
+func newOptions(ioStreams util.IOStreams, kind Kind) *options {
+	return &options{
+		Options: base.Options{
+			IOStreams: ioStreams,
+			Output:    "yaml",
+		},
+		Kind: kind,
+	}
+}
+
+// Complete adapts from the command line args to the data required.
+func (o *options) Complete(f util.Factory, _ *cobra.Command, _ []string) error {
+	manager, err := f.Manager()
+	if err != nil {
+		return err
+	}
+
+	currentTarget, err := manager.CurrentTarget()
+	if err != nil {
+		return err
+	}
+
+	if currentTarget.GardenName() == "" {
+		return target.ErrNoGardenTargeted
+	}
+
+	o.CurrentTarget = currentTarget
+
+	garden, err := manager.Configuration().Garden(currentTarget.GardenName())
+	if err != nil {
+		return err
+	}
+
+	o.Garden = garden
+
+	gardenClient, err := manager.GardenClient(currentTarget.GardenName())
+	if err != nil {
+		return err
+	}
+
+	o.GardenClient = gardenClient
+
+	return nil
+}
+
+// Validate validates the provided command options.
+func (o *options) Validate() error {
+	if o.Options.Output == "" {
+		return errors.New("output must be 'yaml' or 'json'")
+	}
+
+	return nil
+}
+
+// Run does the actual work of the command.
+func (o *options) Run(f util.Factory) error {
+	ctx := f.Context()
+
+	resolvedTarget := ResolvedTarget{
+		Garden: Garden{
+			Name:  o.Garden.Name,
+			Alias: o.Garden.Alias,
+		},
+	}
+
+	if o.Kind == KindGarden {
+		return o.PrintObject(resolvedTarget)
+	}
+
+	if o.CurrentTarget.ProjectName() != "" && o.Kind == KindProject {
+		project, err := o.GardenClient.GetProject(ctx, o.CurrentTarget.ProjectName())
+		if err != nil {
+			return err
+		}
+
+		resolvedTarget.Project = &Project{
+			Name:      project.Name,
+			Namespace: project.Spec.Namespace,
+		}
+
+		return o.PrintObject(resolvedTarget)
+	}
+
+	if o.CurrentTarget.SeedName() != "" && o.Kind == KindSeed {
+		// We already have the seed name, however we get the seed in order to verify that it exists
+		seed, err := o.GardenClient.GetSeed(ctx, o.CurrentTarget.SeedName())
+		if err != nil {
+			return err
+		}
+
+		resolvedTarget.Seed = &Seed{
+			Name: seed.Name,
+		}
+
+		return o.PrintObject(resolvedTarget)
+	}
+
+	shoot, err := findShoot(ctx, o.GardenClient, o.CurrentTarget)
+	if err != nil {
+		if errors.Is(err, target.ErrNoShootTargeted) && o.Kind == KindProject {
+			return target.ErrNoProjectTargeted
+		}
```

Comment:
```suggestion
		if errors.Is(err, target.ErrNoShootTargeted) {
		         if o.Kind == KindProject {
		                 return target.ErrNoProjectTargeted
		         }
		         if o.Kind == KindSeed {
		                 return target.ErrNoSeedTargeted
		         }
		}
```
I no project or seed or shoot is targeted I get the following errors:
```sh
$ g resolve seed
Error: no shoot targeted
```
but
```sh
$ g resolve project
Error: no project targeted
```
Is there a reason to tweak the error for projects but not for seeds?

Reply:
✅  I also added a small test